worlds/witness/player_items.py

```diff
@@ -148,39 +149,119 @@ def get_filler_items(self, quantity: int) -> Dict[str, int]:
-                    else:
-                        # Assume this is some other kind of iterable.
-                        for inner_item in plando_setting[item_setting_key]:
-                            if isinstance(inner_item, str):
-                                output -= {inner_item}
-                            elif isinstance(inner_item, dict):
-                                output -= {item for item, weight in inner_item.items() if weight}
-
-        # Sort the output for consistency across versions if the implementation changes but the logic does not.
-        return sorted(output)
+        output: Dict[str, List[str]] = {}
+
+        existing_items_lookup = set(existing_items)
+
+        if "Symbol" in self._world.options.early_good_items.value:
+            good_symbols = ["Dots", "Black/White Squares", "Symmetry", "Shapers", "Stars"]
+
+            if self._world.options.shuffle_discarded_panels:
+                if self._world.options.puzzle_randomization == "sigma_expert":
+                    good_symbols.append("Arrows")
+                else:
+                    good_symbols.append("Triangles")
+
+            # Replace progressive items with their parents.
+            good_symbols = [
+                static_witness_logic.get_parent_progressive_item(item) for item in good_symbols
+            ]
+
+            output["Symbol"] = [symbol for symbol in good_symbols if symbol in existing_items_lookup]
+
+        if "Door / Door Panel" in self._world.options.early_good_items.value:
+            good_doors = [
+                "Desert Doors & Elevator", "Keep Hedge Maze Doors", "Keep Pressure Plates Doors",
+                "Shadows Lower Doors", "Tunnels Doors",
+
+                "Keep Tower Shortcut (Door)", "Shadows Timed Door", "Tunnels Town Shortcut (Door)",
+
+                "Desert Panels", "Keep Hedge Maze Panels",
+
+                "Shadows Door Timer (Panel)", "Keep Hedge Maze 1 (Panel)", "Town Maze Stairs (Panel)",
+            ]
+
+            # While Caves Shortcuts don't unlock anything in symbol shuffle, you'd still rather have them early.
+            # But, we need to do some special handling for them.
+
+            if self._world.options.shuffle_doors in ("doors", "mixed"):
+                # Caves Shortcuts might exist in vanilla/panel doors because of "early caves: add to pool".
+                # But if the player wanted them early, they would have chosen "early caves: starting inventory".
+                # This is why we make sure these are "natural" Caves Shortcuts.
+                good_doors.append("Caves Shortcuts")
+
+            # These two doors are logically equivalent, so we choose a random one as to not give them twice the chance.
+            good_doors.append(
+                self._world.random.choice(["Caves Mountain Shortcut (Door)", "Caves Swamp Shortcut (Door)"])
+            )
+
+            if self._world.options.shuffle_vault_boxes:
```

Comment:
Shouldn't this require non-randomized as well?

Reply:
Tru